src/System/Console/Hawk/Args/Parse.hs

```diff
@@ -109,11 +107,25 @@ inputSpec (rSep, fSep) = InputSpec <$> source <*> format
 -- >>> test ["-o\t", "-d,", "-O|"]
 -- UseStdout
 -- ("|","\t")
+--
+-- TODO: write test cases for in-place edit
 outputSpec :: (Functor m, Monad m)
            => CommonSeparators -> OptionParserT HawkOption m OutputSpec
-outputSpec (r, f) = OutputSpec <$> sink <*> format
+outputSpec (r, f, file) = OutputSpec <$> sink <*> format
   where
-    sink = return UseStdout
+    sink = do
+        backupSuffix <- consumeLast Option.InPlaceEdit "" $
+                            consumeNullable "<none>" consumeString
+        if null file
+            then if null backupSuffix
+                then return UseStdout
+                else fail "in-place edit requires input file"
+            else if null backupSuffix
+                then return UseStdout
+                else if backupSuffix == "<none>"
```

Comment:
Why not `""`?

Reply:
As the consumeNullable just returns a string, we do not know the whether the argument "-i" was provided or not. So a default value is used to check whether "-i" was provided with no backup file.

Reply:
Yes, but why use "<none>" for this default value? The value "<none> looks like a legal suffix whereas "" does not, so it would make more sense to me if "" was used as the default.

Even better: I'll see if I can refactor this weird nullable API to return a Maybe instead of forcing a default.

Reply:
Ah, I finally get it: you're not using `""` because `""` represents the case in which the `--in-place` option was not given, while `"<none>"` represents the case in which the `--in-place` was given with no suffix